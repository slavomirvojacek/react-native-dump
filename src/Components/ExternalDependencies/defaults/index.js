import { noop } from '@openmaths/utils'

const props = {
  logIntoConsole: noop,
}

export {
  props
}
