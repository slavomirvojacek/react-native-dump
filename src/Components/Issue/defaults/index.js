const issueDefault = (override = {}) => {
  const def = {
    id: '',
    title: '',
    url: '',
    bodyText: '',
    reactions: {
      nodes: []
    }
  }

  return {...def, ...override}
}

const props = {
  issue: issueDefault(),
}

export {
  issueDefault, props
}
