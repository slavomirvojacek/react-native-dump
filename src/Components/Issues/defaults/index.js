const props = {
  issues: [],
}

export {
  props
}
