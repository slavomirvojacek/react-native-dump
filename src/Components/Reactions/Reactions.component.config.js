// import { None } from '@threestup/monads'
import PropTypes from 'prop-types'

const displayName = 'IssuesComponent'

const propTypes = {
  me: PropTypes.object.isRequired,
  reactions: PropTypes.object.isRequired,
}

// Only used for testing
// const testDefaultProps = {
//   me: None,
//   reactions: None,
// }

export default {
  displayName, propTypes
}
