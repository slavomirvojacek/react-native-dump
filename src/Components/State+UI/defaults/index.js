const props = {
  color: 'black', // has to be a valid colour, otherwise warnings are thrown
  isWarningVisible: false,
}

const state = {
  ...props,
  color: 'green', // has to be a valid colour, otherwise warnings are thrown
}


export {
  props, state
}
