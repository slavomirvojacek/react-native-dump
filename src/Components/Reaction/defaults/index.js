import { noop } from '@openmaths/utils'

import { issueDefault } from '../../Issue/defaults'
import { ReactionType } from '../../../Constants/ReactionType'

const userDefault = (override = {}) => {
  const def = {
    id: '',
  }

  return {...def, ...override}
}

const reactionDefault = (override = {}) => {
  const def = {
    id: '',
    content: '',
    user: userDefault()
  }

  return {...def, ...override}
}

const props = {
  type: ReactionType.DEFAULT,
  addReactionToIssue: noop,
  removeReactionFromIssue: noop,
  pressed: false,
  issue: issueDefault()
}

export {
  userDefault, reactionDefault, props
}
