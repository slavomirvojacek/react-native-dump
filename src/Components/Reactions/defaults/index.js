import { None } from '@threestup/monads'

import { issueDefault } from '../../Issue/defaults'

const props = {
  issue: issueDefault(),
  me: None
}

export {
  props
}
