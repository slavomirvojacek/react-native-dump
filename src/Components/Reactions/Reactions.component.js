import { throwIfNotObject } from '@openmaths/utils'
import { Some, None, get_in } from '@threestup/monads'
import React from 'react'
import { View } from 'react-native'

import Config from './Reactions.component.config'
import styles from './Reactions.component.styles'

import Reaction from '../Reaction/Reaction.component'
import { ReactionType, AllowedReactionTypes } from '../../Constants/ReactionType'

class Reactions extends React.Component {
  static hasMyReaction = (reactions = None, type = ReactionType.DEFAULT, me = None) => {
    const myUserId = me.map(_ => _.id).unwrap_or('')

    return reactions
      .and_then(_ => Some(_.find(r => r.content === type)))
      .and_then(_ => get_in(_, 'user.id'))
      .match({
        some: _ => (_ === myUserId),
        none: false
      })
  }

  render () {
    const {issue, me = None, ...rest} = this.props

    throwIfNotObject(issue)

    const reactions = get_in(issue, 'reactions.nodes')

    // To make React keys happy
    const uniqueStr = Math
      .random()
      .toString()

    return (
      <View style={styles.wrapper}>
        {AllowedReactionTypes
          .map((type) => (
            <Reaction
              key={`${type}_${uniqueStr}`}
              pressed={Reactions.hasMyReaction(reactions, type, me)}
              type={type}
              {...rest} />
          ))}
      </View>
    )
  }
}

Reactions.displayName = Config.displayName
Reactions.propTypes = Config.propTypes

export { ReactionType, AllowedReactionTypes }
export default Reactions

