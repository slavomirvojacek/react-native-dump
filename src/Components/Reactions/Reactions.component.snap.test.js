import { Some } from '@threestup/monads'

import Reactions from './Reactions.component'

import { props as testDefaultProps } from './defaults'
import { issueDefault } from '../Issue/defaults'
import { reactionDefault, userDefault } from '../Reaction/defaults'
import { ReactionType } from '../../Constants/ReactionType'
import { assertSnapshots } from '../../TestUtils'

jest
  .mock('../Reaction/Reaction.component', () => 'Reaction')

describe('Reactions Snapshots', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  const configs = [
    {
      props: {...testDefaultProps},
      desc: 'renders correctly',
    },
    {
      props: {
        ...testDefaultProps,
        issue: issueDefault({
          reactions: {
            nodes: [reactionDefault(), reactionDefault()]
          }
        }),
      },
      desc: 'renders correctly when reactions present',
    },
    {
      props: {
        ...testDefaultProps,
        me: Some(userDefault()),
        issue: issueDefault({
          reactions: {
            nodes: [reactionDefault(), reactionDefault()]
          }
        }),
      },
      desc: 'renders correctly when user present and reactions present',
    },
    {
      props: {
        ...testDefaultProps,
        me: Some(userDefault({id: 'someId'})),
        issue: issueDefault({
          reactions: {
            nodes: [
              reactionDefault({
                content: ReactionType.THUMBS_UP,
                user: userDefault({id: 'someId'})
              }),
              reactionDefault(),
              reactionDefault()
            ]
          }
        }),
      },
      desc: 'renders correctly when user present and reactions present with matching type and userId',
    },
  ]

  assertSnapshots(Reactions, configs)
})
