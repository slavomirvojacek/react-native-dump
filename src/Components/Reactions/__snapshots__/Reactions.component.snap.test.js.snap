// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reactions Snapshots renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Reaction
    pressed={false}
    type="THUMBS_UP"
  />
  <Reaction
    pressed={false}
    type="THUMBS_DOWN"
  />
</View>
`;

exports[`Reactions Snapshots renders correctly when reactions present 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Reaction
    pressed={false}
    type="THUMBS_UP"
  />
  <Reaction
    pressed={false}
    type="THUMBS_DOWN"
  />
</View>
`;

exports[`Reactions Snapshots renders correctly when user present and reactions present 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Reaction
    pressed={false}
    type="THUMBS_UP"
  />
  <Reaction
    pressed={false}
    type="THUMBS_DOWN"
  />
</View>
`;

exports[`Reactions Snapshots renders correctly when user present and reactions present with matching type and userId 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Reaction
    pressed={true}
    type="THUMBS_UP"
  />
  <Reaction
    pressed={false}
    type="THUMBS_DOWN"
  />
</View>
`;
